src/T2DMap.cpp

```diff
@@ -4628,10 +4657,10 @@ void T2DMap::wheelEvent(QWheelEvent* e)
     return;
 }
 
-void T2DMap::setMapZoom(int zoom)
+void T2DMap::setMapZoom(qreal zoom)
 {
-    int oldZoom = xyzoom;
-    xyzoom = qMax(3, zoom);
+    qreal oldZoom = xyzoom;
+    xyzoom = qMax(3.0, zoom);
     if (oldZoom != xyzoom) {
```

Comment:
Another equality/inequality test on floating point numbers!

Reply:
Well, yes. However, float comparisons are dangerous only if you have fractional values that may end up close to each other. Here we're using 3.0 as a hard boundary (exact value), otherwise `oldZoom` and `xyzoom` will always be either significantly different or identical (multiplying with 1.0 never changes a float), and even if zooming back in ends up at 3.000001 the single additional step down to 3.0 doesn't hurt anybody.

Thus I didn't bother to look up whatever float-almost-equality method Qt has. Though I can certainly add it ;-)